core/common/src/main/java/alluxio/recorder/NoopRecorder.java

```diff
@@ -19,11 +19,13 @@
  */
 public class NoopRecorder extends Recorder {
   private final LinkedList<String> mRecords;
+  static final Recorder INSTANCE = new NoopRecorder();
 
   /**
    * Constructs a new instance.
    */
-  public NoopRecorder() {
+  private NoopRecorder() {
+    super();
     mRecords = new LinkedList<>();
```

Comment:
let's use ArrayList for better locality https://stackoverflow.com/a/52885253/4933827 the extra cost of copy is trivial

Reply:
I stick to `LinkedList` as it allows clearing the recorded messages and freeing memory.